### PR TITLE
docs/development/verilog_modules.md: new file

### PR DESCRIPTION
**Pull request by krystian-hebel**

### CODE REVIEW

**Comment by miczyg1** [comment]
Looks fine, waiting for TBDs to be filled

File: docs/development/verilog_modules.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  It should never be driven if `lpc_data_wr` is also active. `lpc_data_rd` is
+  sampled by LPC module on falling `clk_i` edge. Data provider module may either
+  start driving both `lpc_data_rd` and `lpc_data_io` signals on rising edge, or
+  drive `lpc_data_io` for at least half a cycle (15 ns) before asserting
+  `lpc_data_rd`. This signal should be driven until LPC module stops driving
+  `lpc_data_req`, after which this signal should be stopped being driven no
+  later than 1 `clk_i` cycle (30ns). LPC module changes `lpc_data_req` on
+  falling `clk_i` edge so it is suggested that data provider module samples that
+  signal on rising edge.
+- `input irq_num`: IRQ number, for TPM this is `TPM_INT_VECTOR_x.sirqVec`. No
```
Comment (miczyg1) [inline docs/development/verilog_modules.md:109]: Maybe add info what TPM_INT_VECTOR_x means (link to TCG spec?)

File: docs/development/verilog_modules.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  parsing is done by LPC module, meaning that both IRQ0 (in some parts of TPM
+  specification this means "IRQ disabled") and IRQ2 (SMI) are valid. IRQ number
+  is sampled on `clk_i` falling edge on SERIRQ start frame, turn-around phase.
+  This is done to avoid sending one interrupt with two different IRQ numbers in
+  one cycle.
+- `input interrupt`: whether interrupt should be signaled to host, active high.
```
Comment (miczyg1) [inline docs/development/verilog_modules.md:115]: What host? LPC host? Host as data provider?

File: docs/development/verilog_modules.md (hunk 1 above)
Comment (miczyg1) [inline docs/development/verilog_modules.md:109]: Explain IRQ - Interrupt request

File: docs/development/verilog_modules.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  drive `lpc_data_io` for at least half a cycle (15 ns) before asserting
+  `lpc_data_rd`. This signal should be driven until LPC module stops driving
+  `lpc_data_req`, after which this signal should be stopped being driven no
+  later than 1 `clk_i` cycle (30ns). LPC module changes `lpc_data_req` on
+  falling `clk_i` edge so it is suggested that data provider module samples that
+  signal on rising edge.
+- `input irq_num`: IRQ number, for TPM this is `TPM_INT_VECTOR_x.sirqVec`. No
+  parsing is done by LPC module, meaning that both IRQ0 (in some parts of TPM
+  specification this means "IRQ disabled") and IRQ2 (SMI) are valid. IRQ number
+  is sampled on `clk_i` falling edge on SERIRQ start frame, turn-around phase.
```
Comment (miczyg1) [inline docs/development/verilog_modules.md:112]: SERIRQ start frame? An explanation would be required if we mention SERIRQ. SOme links to SERIRQ specs?

File: docs/development/verilog_modules.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  This is done to avoid sending one interrupt with two different IRQ numbers in
+  one cycle.
+- `input interrupt`: whether interrupt should be signaled to host, active high.
+  It is checked at the beginning of IRQ frame of IRQ latched from `irq_num`,
+  both in quiet and continuous mode. In addition to that, in quiet mode this
```
Comment (miczyg1) [inline docs/development/verilog_modules.md:117]: Again SERIRQ quiet and continuous mode should be irrelevant to data provider, yet we introduce those without explanation or links to SERIRQ specs

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/development/verilog_modules.md (hunk 1 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: Added

File: docs/development/verilog_modules.md (hunk 2 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: I've described it more, see if it is good now

File: docs/development/verilog_modules.md (hunk 1 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: Done

File: docs/development/verilog_modules.md (hunk 3 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: I can't find any public, official SERIRQ specifications.

Comment (miczyg1) [reply docs/development/verilog_modules.md]: Now you have it: https://web.archive.org/web/20150502224201/http://hackipedia.org/Platform/x86/PCI/Serialized%20IRQ%20Support%20for%20PCI%20Systems.doc.pdf
Source: https://en.wikipedia.org/wiki/Low_Pin_Count#External_links

File: docs/development/verilog_modules.md (hunk 4 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: Link added.

File: docs/development/verilog_modules.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Ports for LPC interface (refer to [LPC specification](https://www.intel.com/content/dam/www/program/design/us/en/documents/low-pin-count-interface-specification.pdf)
+for details):
+
+- `input clk_i`: LPC clock.
+- `input nrst_i`: LPC reset (active low).
+- `input lframe_i`: LPC frame input (active low).
+- `inout lad_bus`: LPC data bus, slow pull-up on host side.
+- `inout serirq`: LPC SERIRQ signal, slow pull-up on host side. More information
+  about SERIRQ can be found [here](https://web.archive.org/web/20150502224201/http://hackipedia.org/Platform/x86/PCI/Serialized%20IRQ%20Support%20for%20PCI%20Systems.doc.pdf)
+  or in PCI specifications (not publicly available).
```
Comment (miczyg1) [inline docs/development/verilog_modules.md:78]: which PCI specification? I couldn;t find SERIRQ mentions in PCI local bus specification nor PCI to PCI bridge architecture specification

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/development/verilog_modules.md (hunk 5 above)
Comment (krystian-hebel) [reply docs/development/verilog_modules.md]: I can't find it anymore. Removed that part.

**Review by miczyg1 — APPROVED** [review]

**Comment by miczyg1** [comment]
Approved, but the pre-commit fails.

**Comment by krystian-hebel** [comment]
I've fixed `mkdocs.yml` on which pre-commit was complaining, now it wants me to add output to `cd` commands, at the same time complaining that the output of device utilization contains tabs instead of spaces. Output of some commands is several screens long. I'd rather not add it all...

It also doesn't like link (`docs/tutorials/building.md:131:12 MD051/link-fragments Link fragments should be valid [Context: "[the easy path](#building-easy)"]`) which I had to change to actually work in pages produced by mkdocs. Previous version of that link which works for GFM (`#building---easy`) doesn't work with mkdocs. Not sure if this is configurable, but in this case it is false-positive. @macpijan CC

**Comment by macpijan** [comment]
@krystian-hebel 

> now it wants me to add output to cd command

No, it wants tou to remove the `$` where it is not necessary, so it is easier to copy commands from documentation. See: https://github.com/updownpress/markdown-lint/blob/master/rules/014-commands-show-output.md

**Comment by macpijan** [comment]
Hmm, I can see that we do not have the lint config commited in this repo, as we have in: https://github.com/Dasharo/docs/blob/master/.markdownlint.yaml
Which means we use default settings

**Comment by macpijan** [comment]
@krystian-hebel I pushed slightly modified linter config to main branch, so you can rebase.

**Pull request merged**